pythonAPIClient/client.py

```diff
@@ -45,25 +45,149 @@ def __init__(self, clientId, clientSecret):
 
     # TODO: return clientID and client secret
     def register(self, owner, permissionLevel='user'):
-        pass
+        url = Client.base_URL + 'apps/register/?owner=' + owner + \
+              '&permissionLevel=' + permissionLevel
+        header = self.header_builder()
+        response = requests.post(url, headers=header)
+        if response.status_code == 404:
+            raise ResourceNotFoundError(response.json()['message'])
+        elif response.status_code == 422:
+            raise FormatError(response.json()['message'])
+        elif response.status_code == 401:
+            self.request_token()
+            header = self.header_builder()
+            response = requests.post(url, headers=header)
+        elif response.status_code == 500:
+            raise InternalError()
+        return response.json()['clientID'], response.json()['clientSecret']
+
 
     # TODO: update client's owner
     def update_owner(self, clientID, owner):
-        pass
+        """
+        Parameters from path
+        ----------
+        clientID : str
+            Client Id of the application.
+
+        Parameters from query
+        ----------
```

Comment:
the `Parameter` in the docstring is part of format, we should not change that.  `Parameters` are used to tell sphinx the method parameters are defined here. the line below `Parameters` or `Returns` etc, should be at least same length as the word itself

Reply:
also, the python wrapper will strip away the need for users to know or care which request parameter goes to where. for them, parameters that will be put in request url are no different than those put in the query string, to users they are all simply parameters for this python method